mypy/typeanal.py

```diff
@@ -641,8 +649,23 @@ def analyze_callable_type(self, t: UnboundType) -> Type:
                                    fallback=fallback,
                                    is_ellipsis_args=True)
             else:
-                self.fail('The first argument to Callable must be a list of types or "..."', t)
-                return AnyType(TypeOfAny.from_error)
+                try:
+                    if not isinstance(callable_args, UnboundType):
+                        raise ValueError
+                    sym = self.lookup_qualified(callable_args.name, callable_args)
+                    if sym is None:
+                        raise ValueError
+                    tvar_def = self.tvar_scope.get_binding(sym)
+                    if not isinstance(tvar_def, ParamSpecDef):
+                        raise ValueError
+
+                    # TODO(shantanu): construct correct type for paramspec
+                    return AnyType(TypeOfAny.from_error)
+                except ValueError:
```

Comment:
I don't like using exceptions for control flow. Some of the other called functions might rise a ValueError, resulting in a confusing error message. I'd suggest using nested if statements or a helper function (return on error).